test/pkg/firewall/firewall.go

```diff
@@ -0,0 +1,165 @@
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			log.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	// save the nftabes on  /host/etc/nftables/firewall.nft
+	_, err = runBashCommandOnPod(debugPod, fmt.Sprintf("echo '%s' > /host/etc/nftables/firewall.nft", string(NFTtable)), false, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to save rule set to file on node %s: %w", nodeName, err)
+	}
+
+	// run apply nft command for the rules
+	_, err = runBashCommandOnPod(debugPod, "nft -f /etc/nftables/firewall.nft", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to apply rule set on node %s: %w", nodeName, err)
+	}
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	// edit the sys conf to make sure it will be on the list after reboot
+	err = editNftablesConf(debugPod, utilsHelpers)
+	if err != nil {
+		return err
+	}
+
+	_, err = runBashCommandOnPod(debugPod, "/usr/sbin/nft list ruleset > /etc/nftables.conf", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", nodeName, err)
+	}
+
+	_, err = runBashCommandOnPod(debugPod, "systemctl enable nftables", true, utilsHelpers)
+	if err != nil {
+		return fmt.Errorf("failed to enable nftables on node %s: %w", nodeName, err)
+	}
+
+	err = saveNFTablesRules("commatrix", "nftables-"+nodeName, string(output))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func RulesList(nodeName string, utilsHelpers utils.UtilsInterface) ([]byte, error) {
+	debugPod, err := utilsHelpers.CreatePodOnNode(nodeName, consts.DefaultDebugNamespace, consts.DefaultDebugPodImage)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create debug pod on node %s: %w", nodeName, err)
+	}
+
+	defer func() {
+		err := utilsHelpers.DeletePod(debugPod)
+		if err != nil {
+			log.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+		}
+	}()
+
+	output, err := nftList(debugPod, utilsHelpers)
+	if err != nil {
+		return nil, err
+	}
+
+	err = saveNFTablesRules("commatrix-after-reboot", "nftables-"+nodeName, string(output))
+	if err != nil {
+		return nil, err
+	}
+
+	return output, nil
+}
+
+func runBashCommandOnPod(debugPod *v1.Pod, command string, chrootDir bool, utilsHelpers utils.UtilsInterface) ([]byte, error) {
+	var fullCommand string
+	if chrootDir {
+		// Format the command for chroot
+		fullCommand = fmt.Sprintf("chroot %s /bin/bash -c '%s'", "/host", command)
+	} else {
+		fullCommand = command
+	}
+
+	output, err := utilsHelpers.RunCommandOnPod(debugPod, []string{"bash", "-c", fullCommand})
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute command '%s' on node %s: %w", command, debugPod.Spec.NodeName, err)
+	}
+	return output, nil
+}
+
+// Make sure that the icnlude of the new nftables is not there.
+// Then if it not there just add that include.
+func editNftablesConf(debugPod *v1.Pod, utilsHelpers utils.UtilsInterface) error {
```

Comment:
So it's validating something or editing something? I'm not I understand the comment or this functionality and why it's needed